gobblin-service/src/main/java/org/apache/gobblin/service/modules/spec/JobExecutionPlan.java

```diff
@@ -61,6 +63,7 @@
   private ExecutionStatus executionStatus = ExecutionStatus.$UNKNOWN;
   private final int maxAttempts;
   private int currentAttempts = 0;
+  private Optional<Future> jobFuture = Optional.absent();
```

Comment:
Should jobFuture be Optional of type CompletableFuture?

Reply:
SpecProducer interface just mentions Future...
though many producer's return CompletedFuture.
Note that CompletedFuture is different that CompletableFuture.